engine/runtime/src/main/scala/org/enso/interpreter/runtime/IrToTruffle.scala

```diff
@@ -1075,9 +1075,10 @@ class IrToTruffle(
       ir match {
         case _: Expression.Binding =>
         case _ =>
-          val types = ir.getMetadata(TypeSignatures)
+          val types: Option[TypeSignatures.Signature] = ir.getMetadata(TypeSignatures)
           types.foreach { tpe =>
-            val checkNode = extractAscribedType(null, tpe.signature);
+            // TODO ?
```

Comment:
What remains to be done?

Reply:
The PR is still in draft. I'm trying to improve the type error message for returns.

Currently, the error is
```
Type error: expected `expression` to be Integer, but got Text.
```
but that is not really telling the user too much. I want to make it:
```
Type error: expected `function_name` result to be Integer, but got Text.
```
to be more precise.

The `comment` was a simple way of achieving that (adding the function name / context to the type ascription as this comment). I'm not sure if this is the best way to do this, but it seemed the simplest and I wanted to at least do a PoC of this approach.